Tests/SonarQube.Common.UnitTests/CmdLineArgsPropertiesProviderTests.cs

```diff
@@ -0,0 +1,188 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="CmdLineArgsPropertiesProviderTests.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
```

Comment:
These tests are modified versions of the ArgumentProcessor tests i.e. they use the same test cases.
I'll remove the redundant ArgumentProcessor tests in a later commit when I update the ArgumentProcessor to use the new providers.